src/include/sof/schedule/ll_schedule.h

```diff
@@ -30,6 +30,8 @@ extern struct tr_ctx ll_tr;
 
```

Comment:
@iuliana-prodan the patch looks good to me. I understand the reasons behind it. Is it possible at our current level of understanding to specify what is the difference between synchronous and full_sync?

Reply:
@iuliana-prodan also can you enhance the commit message in order to better describe what is the current problem.

This 

``We do this because for the DMA_DOMAIN there are no guarantees
for the accuracy of the period of the registrable task
which drives the entire scheduling``` is fine.

We also need to include in the commit message along the lines the scenario we are using and facing issues.

e.g 

In the case we are using a mixer topology there will be 3 pipelines for playback scenarion
- 1 pipeline for pcm1.0  (task 0)
- 1 pipeline for pcm1.1 (task 1)
- 1 pipeline for DAI (task 2)

Each of these pipelines correspond to a task. A

As far as I understood one of the task0 or task1 sometimes starves thus resulting in timeouts on Host.

Reply:
ack, I'm not fully following the difference between DMA and full sync.

Reply:
@dbaluta @lgirdwood The difference is that now, the DMA synchronous is still time dependent: the task->start is used to schedule a task.
The full synchronous is not using the time anymore - is based on the period of each task.

I could have put all these changes on synchronous attribute, but I wanted it to have a small impact on the other targets that use dma_multi_chan domain.

Reply:
Here’s an example of the issue we see on imx. 
In the case of a mixer topology there will be 3 pipelines for playback scenario:
- 1 pipeline for pcm1.0 (task 0);
- 1 pipeline for pcm1.1 (task 1);
- 1 pipeline for DAI (task 2).

Task 2 is the registrable task and this is always scheduled. 
The other tasks (task 0 and task 1) are scheduled based on task->start. 

The problem we see is that, at some point, the DMA transfer from DAI is finished earlier than task->start for one of the task 0 or task 1 and this is not scheduled anymore (based on the check from https://github.com/thesofproject/sof/blob/master/src/schedule/dma_multi_chan_domain.c#L308).
And, task 2 ends up depleting one of the source buffers, thus blocking the entire data flow.

So, task 0 and task 1 should be run at the frequency they were configured with.

@dbaluta I'll add a part of this in the commit message, as you suggested.

Reply:
@iuliana-prodan thanks got you now. Can we have this as scheduling type for topology too as a subsequent PR as it seems something we should be specifying at that level along side the other pipeline scheduling config.

Reply:
@lgirdwood I've created #3817 for your proposal. Let's continue the discussion there. Thanks!